packages/server/src/routes/cron/notifyOnUnminted.ts

```diff
@@ -1,29 +1,134 @@
-import { Firestore, CollectionReference } from "@google-cloud/firestore";
-import { createApiRoute } from "..";
+import {
+  Firestore,
+  CollectionReference,
+  DocumentSnapshot
+} from "@google-cloud/firestore";
+import { auth as adminAuth, messaging as adminMessaging } from "firebase-admin";
+import { PhoneNumberUtil } from "google-libphonenumber";
+import * as moment from "moment";
+
 import { CronNotifyOnUnmintedApiEndpoint } from "@raha/api-shared/dist/routes/cron/definitions";
+
+import { createApiRoute } from "..";
 import { sendPushNotification } from "../../helpers/sendPushNotification";
-import { messaging as adminMessaging } from "firebase-admin";
+import {
+  calculateMaxMintableForMember,
+  MINT_CAP
+} from "../../helpers/calculateMaxMintableForMember";
+
+// Notify people if it is 12 o'clock in their local time.
+const HOUR_TO_NOTIFY = 12;
 
-const RAHUL_TEST_MEMBER_UID = "CrwYE402nwYPzClqaQNcJxL4QDt1";
+/**
+ * Note on region codes: Google-libphonenumber says they use CIDR region codes.
+ * defaultTimezoneByCountryCode uses ISO-3166-2 region codes. I did a set
+ * intersection on the region codes supported by my default-timezones lib
+ * versus libphonenumber, and the only differences are AC (Ascension Island),
+ * EH (Western Sahara), and TA (Tristan da Cunha). Good enough IMO.
+ */
+function getDefaultOffsetForPhoneNumber(
+  phoneNumber: string
+): string | undefined {
+  const phoneNumberUtil = PhoneNumberUtil.getInstance();
+  const countryCode = phoneNumberUtil.parse(phoneNumber).getCountryCode();
+  if (countryCode) {
+    const regionCode = phoneNumberUtil.getRegionCodeForCountryCode(countryCode);
+    const defaultTimezoneByCountryCode = require("default-timezone-for-country/data/default_timezone_by_country_code.json");
```

Comment:
does this load the file every time the function is called? or does it keep it around?

Reply:
supposedly they're cached: https://stackoverflow.com/questions/9210542/node-js-require-cache-possible-to-invalidate